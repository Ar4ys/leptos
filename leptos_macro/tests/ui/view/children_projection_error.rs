use leptos::*;

#[component]
fn Outer(children: ChildrenFn) -> impl IntoView {
    _ = children;
}

#[component]
fn Inner(children: ChildrenFn) -> impl IntoView {
    _ = children;
}

#[component]
fn Inmost(name: String) -> impl IntoView {
    _ = name;
}

fn main() {
    let name = "Alice".to_string();

    view! {
        <Outer>
            <Inner>
                <Inmost name=name.clone()/>
            </Inner>
        </Outer>
    };
}
