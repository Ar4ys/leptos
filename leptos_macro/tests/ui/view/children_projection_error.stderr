error[E0525]: expected a closure that implements the `Fn` trait, but this closure only implements `FnOnce`
  --> tests/ui/view/children_projection_error.rs:23:13
   |
23 | /             <Inner>
24 | |                 <Inmost name=name.clone()/>
   | |                              ---- closure is `FnOnce` because it moves the variable `name` out of its environment
25 | |             </Inner>
   | |                    ^
   | |                    |
   | |____________________this closure implements `FnOnce`, not `Fn`
   |                      the requirement to implement `Fn` derives from here
   |
   = note: required for `Rc<dyn std::ops::Fn() -> Fragment>` to implement `ToChildren<{closure@$DIR/tests/ui/view/children_projection_error.rs:23:13: 25:21}>`
